Don't use -rpath. It's not needed in MacPorts and causes an error with
older compilers.
--- autosetup/proj.tcl	2025-05-10 01:16:51.000000000 +0800
+++ autosetup/proj.tcl	2025-06-19 18:13:42.000000000 +0800
@@ -956,6 +956,8 @@
 # configure then [get-define libdir] is used (noting that it derives
 # from exec-prefix by default).
 proc proj-check-rpath {} {
+  define LDFLAGS_RPATH ""
+  return 0
   if {[proj-opt-was-provided libdir]
       || [proj-opt-was-provided exec-prefix]} {
     set lp "[get-define libdir]"
